### PR TITLE
Tickets/DM-42466: Add Danish to ts_wep

### PR DESCRIPTION
**Pull request by jfcrenshaw**

### CODE REVIEW

**Comment by jfcrenshaw** [comment]
Jenkins currently failing because Danish is not in the environment. I asked Wouter to add it for us.

File: python/lsst/ts/wep/estimation/danish.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+from lsst.ts.wep.estimation.wfAlgorithm import WfAlgorithm
+from scipy.ndimage import binary_erosion
+from scipy.optimize import least_squares
+
+
+class DanishAlgorithm(WfAlgorithm):
+    """Wavefront estimation algorithm class for Danish.
+
+    The Danish algorithm is based on Janish 2012:
+    http://hdl.handle.net/1721.1/78543
```
Comment (jmeyers314) [inline python/lsst/ts/wep/estimation/danish.py:40]: I think it'd be fair to refer to https://github.com/jmeyers314/danish here too :)

File: python/lsst/ts/wep/estimation/danish.py [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+        from the Zernike fitting. The dict contains entries for "intra"
+        and/or "extra", plus the final zernike estimate under "zk".
+
+        The "intra" and "extra" entries are dictionaries that contain the
+        following entries
+            - "image" - the image that is being fit
+            - "variance" - the background variance that was used for fitting
+            - "zkStart" - the starting Zernike coefficients
+            - "lstsqResult" - dictionary of results returned by least_squares
+            - "zkResid" - the Zernike coefficients fit to the donut
```
Comment (jmeyers314) [inline python/lsst/ts/wep/estimation/danish.py]: Maybe `zkDev` or `zkDeviation`?  `resid` sounds like the offset wrt "truth", which of course we don't know.

File: python/lsst/ts/wep/task/estimateZernikesBase.py [hunk 3]
```diff
@@ -116,6 +123,114 @@ def wfAlgoConfig(self) -> WfAlgorithm:
...
+            zip(donutStampsExtra, donutStampsIntra)
+        ):
+            # Determine and set the defocal offset
+            defocalOffset = np.mean(
+                [
+                    donutExtra.defocal_distance,
+                    donutIntra.defocal_distance,
+                ]
+            )
+            wfEstimator.instrument.defocalOffset = defocalOffset / 1e3
```
Comment (jmeyers314) [inline python/lsst/ts/wep/task/estimateZernikesBase.py]:
```suggestion
            wfEstimator.instrument.defocalOffset = defocalOffset / 1e3  # m -> mm
```

**Review by jmeyers314 — APPROVED** [review]
LGTM!  Just a couple minor comments.

File: python/lsst/ts/wep/estimation/danish.py (hunk 2 above)
Comment (jfcrenshaw) [reply python/lsst/ts/wep/estimation/danish.py]: `zkResid` is also used in [the TIE history](https://github.com/lsst-ts/ts_wep/blob/cad0004cb344479310a995b50d1d4db91e2d9d01/python/lsst/ts/wep/estimation/tie.py#L396). Here residual is meant as "the aberration left over after compensation for `zkStart`". I think it's not a problem to change this to `zkFit` though.

**Pull request merged**